src/main/java/org/jvnet/hudson/test/HudsonTestCase.java

```diff
@@ -1592,11 +1593,25 @@ public void run() {
             return r.get(0);
         }
 
-        public HtmlPage search(String q) throws IOException, SAXException {
+        public HtmlPage search(String q) throws IOException, SAXException, InterruptedException {
             HtmlPage top = goTo("");
-            HtmlForm search = top.getFormByName("search");
-            search.getInputByName("q").setValue(q);
-            return (HtmlPage)HtmlFormUtil.submit(search, null);
+            HtmlButton button = top.querySelector("#button-open-command-palette");
+
+            // Legacy versions of Jenkins
+            if (button == null) {
+                HtmlForm search = top.getFormByName("search");
+                search.getInputByName("q").setValue(q);
+                return (HtmlPage)HtmlFormUtil.submit(search, null);
+            }
+
+            button.click();
+
+            HtmlInput search = top.querySelector("#command-bar");
+            search.setValue(q);
+
+            Thread.sleep(100);
```

Comment:
avoid sleeps and wait for expectedconditions as different speed machines may take different amount of time